### PR TITLE
Add Enum for Display text

### PR DESCRIPTION
**Pull request by akileng56**
- Add new property display text
- Remove the show percentage property

### CODE REVIEW

File: src/ProgressCircle.xml [hunk 1]
```diff
@@ -4,10 +4,15 @@
     <description>Displays a progress in a circle</description>
     <icon>iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAAZdEVYdFNvZnR3YXJlAHBhaW50Lm5ldCA0LjAuMTM0A1t6AAABY0lEQVQ4T6WTvy8EURDHtxCulFCQnCO5QicaqpPrr9pGI+QKjdDQUUk0q9L4Ec0RocDS6a+6lfgXrnMbGnLFKlzs3vp+14y82zg/4iWfZN68N9+dmZ1nxXGc4HlePzgGsYm1h0OD6ZNmvXzVGNU4DS6AhgQFwAE2yC64/pR9/ugUz55uMwdhmyIDh61w9uJhORHAJX5Zg6tgTNXTzLt+YfwoaKoIM6GApl39KijN4vV9Jl95CbQcCmjaXb+cZu7SL/XuRx99EQFHD7F2QAtwMwwmwauwpPdmTp/vTAFbgifAtl4SnwtGxA7Vz8aaAlm5sAUqwrr43jQI6wbkaPPvdBPYAH1gFxTBrwS0hFWwIvYmKIEaGBTfp1i6hKSJWDkQAW4i8ZVBW3xrKsDBMgU6fiPWkNqGr0dtDpROJQX+PEg6jTpI/xtlHvBh0MEDpsb62CR2+sfHpOpUS1JiXd/Q+Zxj6x3u72eCF5jDPgAAAABJRU5ErkJggg==</icon>
     <properties>
-        <property key="showPercentage" type="boolean" defaultValue="true">
-            <caption>Show percentage</caption>
+        <property key="text" type="enumeration" defaultValue="percentage">
```
Comment (acellam) [inline src/ProgressCircle.xml]: i feel this key as `text` is too generic and ambiguous. It makes it hard for another developer to know what its purpose is. I suggest you use another name eg `textDisplayOption` or  `textStyle` , etc

File: src/components/ProgressCircle.ts [hunk 2]
```diff
@@ -16,7 +17,7 @@ export interface ProgressCircleProps {
     onClickAction?: () => void;
     positiveValueColor?: BootstrapStyle;
     style?: object;
-    showPercentage?: boolean;
+    text?: DisplayText;
```
Comment (acellam) [inline src/components/ProgressCircle.ts]: Same as above

File: src/components/ProgressCircle.ts [hunk 3]
```diff
@@ -29,7 +30,7 @@ export class ProgressCircle extends Component<ProgressCircleProps, { alertMessag
     static defaultProps: ProgressCircleProps = {
         animate: true,
         maximumValue: 100,
-        showPercentage: true,
+        text: "percentage",
```
Comment (acellam) [inline src/components/ProgressCircle.ts]: Same as above

File: src/components/ProgressCircle.ts [hunk 4]
```diff
@@ -46,7 +47,7 @@ export class ProgressCircle extends Component<ProgressCircleProps, { alertMessag
 
     componentDidMount() {
         this.createProgressCircle(this.props.circleThickness);
-        this.setProgress(this.props.value, this.props.maximumValue, this.props.showPercentage);
+        this.setProgress(this.props.value, this.props.maximumValue, this.props.text);
```
Comment (acellam) [inline src/components/ProgressCircle.ts]: Same as above

File: src/components/ProgressCircle.ts [hunk 5]
```diff
@@ -57,7 +58,7 @@ export class ProgressCircle extends Component<ProgressCircleProps, { alertMessag
             this.progressCircle.destroy();
             this.createProgressCircle(newProps.circleThickness);
         }
-        this.setProgress(newProps.value, newProps.maximumValue, newProps.showPercentage);
+        this.setProgress(newProps.value, newProps.maximumValue, newProps.text);
```
Comment (acellam) [inline src/components/ProgressCircle.ts]: Same as above

File: src/components/ProgressCircleContainer.ts [hunk 6]
```diff
@@ -19,7 +19,7 @@ export interface ContainerProps extends WrapperProps {
     page?: string;
     progressAttribute: string;
     positiveValueColor: BootstrapStyle;
-    showPercentage: boolean;
+    text: DisplayText;
```
Comment (acellam) [inline src/components/ProgressCircleContainer.ts]: Same as above

File: src/components/__tests__/ProgressCircle.spec.ts [hunk 7]
```diff
@@ -48,7 +48,7 @@ describe("ProgressCircle", () => {
         const setText = progressbar.Circle.prototype.setText as jasmine.Spy;
         spyOnCircle();
 
-        const progress = newCircleInstance({ animate: false, value: 80 });
+        const progress = newCircleInstance({ animate: false, value: 80, text: "percentage" });
```
Comment (acellam) [inline src/components/__tests__/ProgressCircle.spec.ts]: Same as above

File: src/components/__tests__/ProgressCircle.spec.ts [hunk 8]
```diff
@@ -67,91 +67,127 @@ describe("ProgressCircle", () => {
...
 
-        expect(progressCircle.text.textContent).toBe("180%");
-        expect(progressCircle.animate).toHaveBeenCalledWith(1);
-    });
+            expect(progressCircle.text.textContent).toBe("--");
+        });
 
-    it("has the class widget-progress-circle-alert when the maximum value is less than one", () => {
-        const progress = renderProgressCircle({ value: 20, maximumValue: 0 });
+        it("the text set to Invalid when the maximum value is less than 1", () => {
```
Comment (acellam) [inline src/components/__tests__/ProgressCircle.spec.ts:139]: `text is set to invalid when the maximum value is less than 1`

File: src/ProgressCircle.xml (hunk 1 above)
Comment (akileng56) [reply src/ProgressCircle.xml]: Changed

File: src/components/ProgressCircle.ts (hunk 2 above)
Comment (akileng56) [reply src/components/ProgressCircle.ts]: Fixed

File: src/components/ProgressCircle.ts (hunk 3 above)
Comment (akileng56) [reply src/components/ProgressCircle.ts]: Fixed

File: src/components/ProgressCircle.ts (hunk 4 above)
Comment (akileng56) [reply src/components/ProgressCircle.ts]: Fixed

File: src/components/ProgressCircle.ts (hunk 5 above)
Comment (akileng56) [reply src/components/ProgressCircle.ts]: Fixed

File: src/components/ProgressCircleContainer.ts (hunk 6 above)
Comment (akileng56) [reply src/components/ProgressCircleContainer.ts]: Fixed

File: src/components/__tests__/ProgressCircle.spec.ts (hunk 7 above)
Comment (akileng56) [reply src/components/__tests__/ProgressCircle.spec.ts]: Fixed

File: src/components/__tests__/ProgressCircle.spec.ts (hunk 8 above)
Comment (akileng56) [reply src/components/__tests__/ProgressCircle.spec.ts:139]: The test sentence changes if this is changed.
Currently, it reads....ProgressCircle renders a circle with the text set to Invalid when the maximum value is less than 1

File: package.json [hunk 9]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "progresscircle",
   "widgetName": "ProgressCircle",
-  "version": "1.3.0",
+  "version": "1.3.1",
```
Comment (Andries-Smit) [inline package.json]: Its change of feature, should be `1.4`

Comment (akileng56) [reply package.json]: Changed

**Pull request merged**